### PR TITLE
Fix AutoNAT service for private network

### PR DESCRIPTION
**Pull request by requilence**
fixing #6250

### CODE REVIEW

File: core/node/libp2p.go [hunk 1]
```diff
@@ -523,16 +523,19 @@ func P2PPubsubRouter(mctx MetricsCtx, lc fx.Lifecycle, in p2pPSRoutingIn) (p2pRo
...
 
-func AutoNATService(mctx MetricsCtx, lc fx.Lifecycle, cfg *config.Config, host host.Host) error {
+func AutoNATService(repo repo.Repo, mctx MetricsCtx, lc fx.Lifecycle, cfg *config.Config, host host.Host) error {
 	if !cfg.Swarm.EnableAutoNATService {
 		return nil
 	}
-	var opts []libp2p.Option
+
+	// collect private net option in case swarm.key is presented
+	opts, _, _ := P2PPNet(repo)
```
Comment (magik6k) [inline core/node/libp2p.go]: should check the error

Comment (requilence) [reply core/node/libp2p.go]: done

**Comment by requilence** [comment]
> We could try to re-use the protector from the main host, but it's probably not worth it
I was able to do this way in 0.4.20, but with current `fx` refactoring, I'm not sure how to pass it

**Comment by aloknerurkar** [comment]
Does this functionality work as expected? I updated to the latest version of IPFS and I am not able to get the node to do the NAT traversal when I am using a swarm key.

```
LIBP2P_FORCE_PNET=1 ipfs daemon                                                            
Initializing daemon...                                                                                                                         
go-ipfs version: 0.4.23-6ce9a35                                                                                                                
Repo version: 7                                                                                                                                
System version: amd64/darwin                                                                                                                   
Golang version: go1.13.8                                                                                                                       
Swarm is limited to private network of peers with the swarm key                                                                                
Swarm key fingerprint: d4574a64137198e7c6c8366f77e37e46                                                                                        
Swarm listening on /ip4/127.0.0.1/tcp/4001                                                                                                     
Swarm listening on /ip4/192.168.0.98/tcp/4001                                                                                                  
Swarm listening on /ip6/::1/tcp/4001                                                                                                           
Swarm listening on /ip6/fd01::18eb:9ca2:8968:d339/tcp/4001                                                                                     
Swarm listening on /ip6/fd01::39c2:943c:4774:db16/tcp/4001                                                                                     
Swarm listening on /p2p-circuit                                                                                                                
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/192.168.0.98/tcp/4001
Swarm announcing /ip6/::1/tcp/4001
Swarm announcing /ip6/fd01::18eb:9ca2:8968:d339/tcp/4001
Swarm announcing /ip6/fd01::39c2:943c:4774:db16/tcp/4001
API server listening on /ip4/127.0.0.1/tcp/5001
WebUI: http://127.0.0.1:5001/webui
Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready
```

Whereas when I don't use swarm key, I see the NAT traversal happening.
```
ipfs daemon         
Initializing daemon...
go-ipfs version: 0.4.23-6ce9a35
Repo version: 7
System version: amd64/darwin
Golang version: go1.13.8
Swarm listening on /ip4/127.0.0.1/tcp/4001
Swarm listening on /ip4/192.168.0.98/tcp/4001
Swarm listening on /ip6/::1/tcp/4001
Swarm listening on /ip6/fd01::18eb:9ca2:8968:d339/tcp/4001
Swarm listening on /ip6/fd01::39c2:943c:4774:db16/tcp/4001
Swarm listening on /p2p-circuit
**Swarm announcing /ip4/106.51.29.204/tcp/21683**
Swarm announcing /ip4/127.0.0.1/tcp/4001
Swarm announcing /ip4/192.168.0.98/tcp/4001
Swarm announcing /ip6/::1/tcp/4001
Swarm announcing /ip6/fd01::18eb:9ca2:8968:d339/tcp/4001
Swarm announcing /ip6/fd01::39c2:943c:4774:db16/tcp/4001
API server listening on /ip4/127.0.0.1/tcp/5001
WebUI: http://127.0.0.1:5001/webui
Gateway (readonly) server listening on /ip4/127.0.0.1/tcp/8080
Daemon is ready
```

**Comment by Stebalien** [comment]
Please file a new bug report and include your config (minus your private key). Something looks fishy.

**Comment by aloknerurkar** [comment]
Filed https://github.com/ipfs/go-ipfs/issues/7067
Let me know if you need any other information.

I see the autonat-svc used by ipfs is deprecated. Libp2p has a new autonat package that it's using.

**Comment by willscott** [comment]
github.com/libp2p/go-libp2p-autonat-svc/ is now incorporated into github.com/libp2p/go-libp2p-autonat/ which acts as both the service allowing nodes to determine their nat status, and the process by which nodes go about querying other nodes to learn their status.

**Pull request merged**